python/cuml/manifold/umap.pyx

```diff
@@ -272,10 +272,72 @@ class UMAP(Base):
 
         self.umap_params = <size_t > umap_params
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        del state['handle']
+
+        cdef size_t params_t = <size_t>self.umap_params
+        cdef UMAPParams* umap_params = <UMAPParams*>params_t
+
+        state['X_m'] = cudf.DataFrame.from_gpu_matrix(self.X_m)
+        state['arr_embed'] = cudf.DataFrame.from_gpu_matrix(self.arr_embed)
+        state["n_neighbors"] = umap_params.n_neighbors
+        state["n_components"] = umap_params.n_components
+        state["n_epochs"] = umap_params.n_epochs
+        state["learning_rate"] = umap_params.learning_rate
+        state["min_dist"] = umap_params.min_dist
+        state["spread"] = umap_params.spread
+        state["set_op_mix_ratio"] = umap_params.set_op_mix_ratio
+        state["local_connectivity"] = umap_params.local_connectivity
+        state["repulsion_strength"] = umap_params.repulsion_strength
+        state["negative_sample_rate"] = umap_params.negative_sample_rate
+        state["transform_queue_size"] = umap_params.transform_queue_size
+        state["init"] = umap_params.init
+        state["a"] = umap_params.a
```

Comment:
This is sort of where I think that doing `state["params.a"] = params.a` would be a little cleaner, since "a" is such a general term

Reply:
Hmm... maybe this suggestion doesn't work with the self.__dict__.update though

Reply:
I do agree with you, the use of the separate `umap_params` class here is what's complicating this. My comment above might address this.